packages/odyssey-react-mui/src/Badge.tsx

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { CSSProperties, memo, useMemo } from "react";
+
+import {
+  useOdysseyDesignTokens,
+  DesignTokens,
+} from "./OdysseyDesignTokensContext";
+import { Box } from "./Box";
+import type { SeleniumProps } from "./SeleniumProps";
+
+export const badgeTypeValues = ["default", "attention", "danger"] as const;
+
+export type BadgeProps = {
+  badgeContent: number;
+  badgeContentMax?: number;
+  type?: (typeof badgeTypeValues)[number];
+} & SeleniumProps;
+
+const badgeTypeColors = (odysseyTokens: DesignTokens) => ({
+  default: {
+    background: odysseyTokens.HueNeutral200,
+    font: odysseyTokens.TypographyColorBody,
+  },
+  attention: {
+    background: odysseyTokens.PalettePrimaryMain,
+    font: odysseyTokens.TypographyColorInverse,
+  },
+  danger: {
+    background: odysseyTokens.PaletteDangerMain,
+    font: odysseyTokens.TypographyColorInverse,
+  },
+});
+
+const Badge = ({
+  badgeContent,
+  badgeContentMax = 999,
+  type = "default",
+}: BadgeProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const greaterThanZeroContentMax = badgeContentMax > 0 ? badgeContentMax : 1;
+  const threeDigitLimitedMax =
+    greaterThanZeroContentMax > 999 ? 999 : greaterThanZeroContentMax;
+  const isOverContentMax = Boolean(
+    badgeContent && badgeContent > threeDigitLimitedMax
```

Comment:
This won't work if `badgeContent` is `0`

Reply:
I _think_ that is fine. We'd want to hide the badge if the content is `0`.

Reply:
Maybe a test should be added for this if one doesn't exist yet? If `badgeContent = 0`, `isOverContentMax` will be `false` (which seems incorrect to me as 0 is definitely less than 999), and `formattedContent` will be `0`. I think it will render, no? Apologies if I didn't follow the logic correctly, but this is what it seems like it will do.